app/src/pages/trace/SpanDetails.tsx

```diff
@@ -0,0 +1,1667 @@
+  [SemanticAttributePrefixes.tool]?: AttributeTool;
+  [SemanticAttributePrefixes.reranker]?: AttributeReranker;
+  [SemanticAttributePrefixes.llm]?: AttributeLlm;
+};
+
+type Span = Extract<SpanDetailsQuery$data["span"], { __typename: "Span" }>;
+
+type DocumentEvaluation = NonNullable<Span["documentEvaluations"]>[number];
+
+/**
+ * Hook that safely parses a JSON string.
+ */
+const useSafelyParsedJSON = (
+  jsonStr: string
+): { json: { [key: string]: unknown } | null; parseError?: unknown } => {
+  return useMemo(() => {
+    try {
+      return { json: JSON.parse(jsonStr) };
+    } catch (e) {
+      return { json: null, parseError: e };
+    }
+  }, [jsonStr]);
+};
+
+const spanHasException = (span: Span) => {
+  return span.events.some((event) => event.name === "exception");
+};
+
+/**
+ * Card props to apply across all cards
+ */
+const defaultCardProps: Partial<CardProps> = {
+  backgroundColor: "light",
+  borderColor: "light",
+  variant: "compact",
+  collapsible: true,
+};
+
+export function SpanDetails({
+  spanNodeId,
+  projectId,
+}: {
+  /**
+   * The Global ID of the span
+   */
+  spanNodeId: string;
+  projectId: string;
+}) {
+  const { span } = useLazyLoadQuery<SpanDetailsQuery>(
+    graphql`
+      query SpanDetailsQuery($spanId: GlobalID!) {
+        span: node(id: $spanId) {
+          __typename
+          ... on Span {
+            id
+            context {
+              spanId
+              traceId
+            }
+            name
+            spanKind
+            statusCode: propagatedStatusCode
+            statusMessage
+            startTime
+            parentId
+            latencyMs
+            tokenCountTotal
+            tokenCountPrompt
+            tokenCountCompletion
+            input {
+              value
+              mimeType
+            }
+            output {
+              value
+              mimeType
+            }
+            attributes
+            events @required(action: THROW) {
+              name
+              message
+              timestamp
+            }
+            spanEvaluations {
+              name
+              label
+              score
+            }
+            documentRetrievalMetrics {
+              evaluationName
+              ndcg
+              precision
+              hit
+            }
+            documentEvaluations {
+              documentPosition
+              name
+              label
+              score
+              explanation
+            }
+            ...SpanEvaluationsTable_evals
+          }
+        }
+      }
+    `,
+    {
+      spanId: spanNodeId,
+    }
+  );
+
+  if (span.__typename !== "Span") {
+    throw new Error(
+      "Expected a span, but got a different type" + span.__typename
+    );
+  }
+
+  const hasExceptions = useMemo<boolean>(() => {
+    return spanHasException(span);
+  }, [span]);
+  const showAnnotations = useFeatureFlag("annotations");
+  return (
+    <Flex direction="column" flex="1 1 auto" height="100%">
```

Comment:
stuff below here is just copied over from old trace page span details correct?

Reply:
yes